test/tests/Addresses/AddressTypeTest.php

```diff
@@ -6,7 +6,7 @@
 use IPLib\Address\Type;
 use PHPUnit_Framework_TestCase;
 
-class AddressTypeTest extends PHPUnit_Framework_TestCase
+class RangeRangeTypeTest extends PHPUnit_Framework_TestCase
```

Comment:
Why did you change the class name?
Furthermore, let's follow the PSR-4 rules: the class name must be the same as the file name (except for the leading .php file extension).

Reply:
Class name collision caused composer to complain and fail. Tests aren't complete yet as Pattern testing isn't covered, we'll of course lean up all the file names etc when we're at that stage so consider that very much WIP.

Reply:
> Class name collision caused composer to complain and fail

Since we use namespaces (that reflects the folder structure), collisions occur only in case the fully qualified class names (that is, namespace+class, like IPLib\Test\Addresses\RangeTypeTest) are the same.
But since we should follow PSR-4 (the namespace is tied to the folder and the class name is tied to the file name), we shouldn't have collisions... What are the fully qualified names of the classes that collides?

Reply:
https://travis-ci.org/mlocati/ip-lib/jobs/315451142

`Fatal error: Cannot declare class IPLib\Test\Addresses\TypeTest, because the name is already in use in /home/travis/build/mlocati/ip-lib/test/tests/Ranges/RangeTypeTest.php on line 296`

However, I just noticed the Range TypeTest is in the wrong namespace... I blame lack of coffee on that one. I'll get it fixed up.

:)

Reply:
😆 👍